src/core/Akka.Tests/Pattern/BackoffSupervisorSpec.cs

```diff
@@ -372,46 +392,52 @@ IActorRef WaitForChild()
             var c1 = ExpectMsg<BackoffSupervisor.CurrentChild>().Ref;
             Watch(c1);
             c1.Tell(PoisonPill.Instance);
-            ExpectTerminated(c1);
```

Comment:
This can rarely fail, need to track down the real cause because its not supposed to happen.

Reply:
How can that fail? If `c1` processes the `PosionPill` it has to die? Is this taking longer than 3 seconds?

Reply:
No, the test received a Terminated message but from another actor, which is really weird.
It only happened once every 1000 test or so iteration, making it hard to debug.

Reply:
ah ok - so if we're death watching two actors the easiest way to solve that race condition is to create two different test probes, one to death watch each actor. No overlap that way. Although I expect in this case what makes it hard is that it's two different children from the same parent being watched, right?

Reply:
That might be a solution, will have to explore it a bit more